actor-typed-tests/src/test/scala/org/apache/pekko/actor/typed/SupervisionSpec.scala

```diff
@@ -1232,34 +1222,33 @@ class SupervisionSpec extends ScalaTestWithActorTestKit("""
     "work with nested supervisions and defers" in {
       val strategy = SupervisorStrategy.restart.withLimit(3, 1.second)
       val probe = TestProbe[AnyRef]("p")
-      val beh = supervise[String](setup(_ =>
-        supervise[String](setup { _ =>
-          probe.ref ! Started
-          Behaviors.empty[String]
-        }).onFailure[RuntimeException](strategy))).onFailure[Exception](strategy)
+      val behv = supervise[String](setup { _ =>
```

Comment:
Would it make sense to also test the old way too? We should have regression testing of old behavior too.

Reply:
That makes sense, I will add additional tests.

Reply:
the PR https://github.com/apache/pekko/pull/1386 will keep the same behavior as the old way, it is just syntax sugar.

i will make extra test for it, in case we want change it in the future.